service/controller/resource/namespace/current.go

```diff
@@ -71,12 +74,17 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
 	// remove the namespace as usual.
 	if key.IsDeleted(&customObject) {
 		n := key.ClusterNamespace(customObject)
-		list, err := r.k8sClient.CoreV1().Pods(n).List(ctx, metav1.ListOptions{})
+
+		lo := metav1.ListOptions{
+			LabelSelector: fmt.Sprintf("%s=%s", label.ManagedBy, project.Name()),
+		}
+
+		list, err := r.k8sClient.AppsV1().Deployments(n).List(ctx, lo)
 		if err != nil {
 			return nil, microerror.Mask(err)
 		}
 		if len(list.Items) != 0 {
-			r.logger.Debugf(ctx, "cannot finish deletion of namespace due to existing pods")
+			r.logger.Debugf(ctx, "cannot finish deletion of namespace due to existing deployments")
```

Comment:
Changing the log to `deployments`  as well.

Reply:
We drain the pods so does that mean the deployment is not deleted till after the pods?

If so then I think this is fine.

Reply:
Yes, exactly. kvm-operator will be deleting the deployment in the deletion phase. So it's fine.  the cluster deletion phase. https://github.com/giantswarm/kvm-operator/blob/682a07b57f78d1742c27c2c42a858b7eefd77e51/service/controller/resource/deployment/delete.go#L28-L36

Reply:
OK cool thanks for confirming.